src/evaluators/interpolation/PHAL_DOFGradInterpolation_Def.hpp

```diff
@@ -141,6 +141,7 @@ template< typename Traits>
 KOKKOS_INLINE_FUNCTION
 void FastSolutionGradInterpolationBase<PHAL::AlbanyTraits::Jacobian, Traits, typename PHAL::AlbanyTraits::Jacobian::ScalarT>::
 operator() (const FastSolutionGradInterpolationBase_Jacobian_Tag& tag, const int& cell) const {
+  const int num_dof = this->val_node(0,0).size();
   for (size_t qp=0; qp < this->numQPs; ++qp) {
     for (size_t dim=0; dim<this->numDims; dim++) {
       this->grad_val_qp(cell,qp,dim) = ScalarT(num_dof, this->val_node(cell, 0).val() * this->GradBF(cell, 0, qp, dim));
```

Comment:
We can probably optimize this similarly to how Oscar did it in the other PR. This can be done in in a subsequent PR.

Reply:
Oscar is planning to look at other kernels starting with the most expensive ones showing up in the profiler. I think this is in the top 10.